include/particles/particle_bbar_implicit.tcc

```diff
@@ -0,0 +1,213 @@
+// Compute B matrix
+template <>
+inline Eigen::MatrixXd mpm::ParticleBbar<1>::compute_bmatrix() noexcept {
```

Comment:
I think we can write these functions together in `particle_bbar.tcc`. They are not that much anyway.
Also, let's put them into a folder `include/particles/anti_locking/`